packages/hardhat/contracts/Pools.sol

```diff
@@ -0,0 +1,161 @@
+pragma solidity ^0.8.4;
+//SPDX-License-Identifier: MIT
+
+import "hardhat/console.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "prb-math/contracts/PRBMath.sol";
+import "prb-math/contracts/PRBMathUD60x18.sol";
+
+import "./interfaces/IPools.sol";
+import "./interfaces/Validated.sol";
+
+contract Pools is IPools, Validated {
+    using SafeERC20 for ERC20;
+    using PRBMathUD60x18 for uint256;
+
+    // This constant represents the utilization rate at which the pool aims to obtain most competitive borrow rates.
+    uint256 public immutable optimalUtilisationRate;
+    // This constant represents the excess utilization rate above the optimal.
+    // It's always equal to 1-optimal utilization rate. Added as a constant here for gas optimizations.
+    uint256 public immutable excessUtilisationRate;
+    // Base variable borrow rate when Utilization rate = 0.
+    uint256 internal immutable baseBorrowRate;
+    // Slope of the variable interest curve when utilization rate > 0 and <= optimalUtilisationRate.
+    uint256 internal immutable slope1;
+    // Slope of the variable interest curve when utilization rate > optimalUtilisationRate.
+    uint256 internal immutable slope2;
+
+    mapping(address => Pool) private pools;
+
+    constructor (
+        uint256 _optimalUtilisationRate,
+        uint256 _baseBorrowRate,
+        uint256 _slope1,
+        uint256 _slope2
+    ) {
+        optimalUtilisationRate = _optimalUtilisationRate;
+        excessUtilisationRate = PRBMathUD60x18.SCALE - _optimalUtilisationRate;
+        baseBorrowRate = _baseBorrowRate;
+        slope1 = _slope1;
+        slope2 = _slope2;
+    }
+
+    function balance(address token) external view returns (uint256) {
+      return pools[token].balance;
+    }
+
+    function borrowed(address token) external view returns (uint256) {
+      return pools[token].borrowed;
+    }
+
+    function available(address token) external view override returns (uint qty) {
+        qty = pools[token].balance - pools[token].borrowed;
```

Comment:
we should understand the difference (if any) between assigning to a named return var vs just returning, the inconsistency is killing me :P